pkg/deployments/tasks/20220413-polygon-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,182 @@
+
+    expect(await factory.isGaugeFromFactory(gauge.address)).to.be.true;
+    expect(await factory.getRecipientGauge(recipient.address)).to.equal(gauge.address);
+    expect(await factory.getGaugeRecipient(gauge.address)).to.equal(recipient.address);
+  });
+
+  it('grant permissions', async () => {
+    // We need to grant permission to the admin to add the Polygon factory to the GaugeAdder, and also to then add
+    // gauges from said factory to the GaugeController.
+    const govMultisig = await impersonate(GOV_MULTISIG, fp(100));
+
+    const selectors = ['addGaugeFactory', 'addPolygonGauge'].map((method) => gaugeAdder.interface.getSighash(method));
+    await Promise.all(
+      selectors.map(
+        async (selector) =>
+          await authorizer.connect(govMultisig).grantRole(await gaugeAdder.getActionId(selector), admin.address)
+      )
+    );
+
+    // We also need to grant permissions to mint in the gauges, which is done via the Authorizer Adaptor
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(await authorizerAdaptor.getActionId(gauge.interface.getSighash('checkpoint')), admin.address);
+  });
+
+  it('add gauge to gauge controller', async () => {
+    await gaugeAdder.addGaugeFactory(factory.address, 3); // Polygon is Gauge Type 3
+    await gaugeAdder.addPolygonGauge(gauge.address);
+
+    expect(await gaugeController.gauge_exists(gauge.address)).to.be.true;
+  });
+
+  it('vote for gauge', async () => {
+    expect(await gaugeController.get_gauge_weight(gauge.address)).to.equal(0);
+    await gaugeController.connect(veBALHolder).vote_for_gauge_weights(gauge.address, 10000); // Max voting power is 10k points
+
+    // We now need to go through an epoch for the votes to be locked in
+    await advanceTime(DAY * 8);
+
+    await gaugeController.checkpoint();
+    expect(await gaugeController['gauge_relative_weight(address)'](gauge.address)).to.be.gt(0);
+  });
+
+  it('mint & bridge tokens', async () => {
+    // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+    // relative weight, as it will change as time goes by due to vote decay.
+    const firstMintWeekTimestamp = await currentWeekTimestamp();
+    const gaugeRelativeWeight = await gaugeController['gauge_relative_weight(address)'](gauge.address);
+
+    const calldata = gauge.interface.encodeFunctionData('checkpoint');
+
+    // Even though the gauge has relative weight, it cannot mint yet as it needs for the epoch to finish
+    const zeroMintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    expectEvent.inIndirectReceipt(await zeroMintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+      periodEmissions: 0,
+    });
+    // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
```

Comment:
```suggestion
    expectEvent.notEmitted(receipt, "Transfer");
```

Reply:
This won't work because the contract we call (the authorizer) is not the one who emits the event: we need the equivalent of `inIndirectReceipt` but for `notEmitted`.

Reply:
(I mean, the assertion will pass, but it won't do the right thing)